yellowbrick/text/dispersion.py

```diff
@@ -79,13 +103,14 @@ def _compute_dispersion(self, text):
                 # In the case that word is not in target words, any empty list is
                 # returned and no data will be yielded
                 self.offset += 1
-                for y in (self.target_words_ == word).nonzero()[0]:
-                    yield (self.offset, y)
+                for y_coord in (self.target_words_ == word).nonzero()[0]:
+                    y_coord = int(y_coord)
+                    yield (self.offset, y_coord, target)
             if self.annotate_docs:
                 self.boundaries_.append(self.offset)
         self.boundaries_ = np.array(self.boundaries_, dtype=int)
 
-    def fit(self, text):
+    def fit(self, text, y=None, **kwargs):
         """
         The fit method is the primary drawing input for the dispersion
         visualization. It requires the corpus as a list of words.
```

Comment:
It looks like there may have been some changes since the original implementation and might need to update the docstring here; does `DispersionPlot` still expect the input data to be a list of words? It looks like it could either be a list of documents or a generator that would yield a list of documents.